internal/characters/lyney/burst.go

```diff
@@ -0,0 +1,121 @@
+package lyney
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/enemy"
+)
+
+var burstFrames []int
+
+const (
+	burstKey           = "lyney-q"
+	burstMarkKey       = "lyney-q-mark"
+	burstStart         = 100
+	burstFirstInterval = 12
+	burstInterval      = 0.15 * 60
+	burstDuration      = 182
+	burstCD            = 15 * 60
+	burstEnergyDelay   = 8
+)
+
+func init() {
+	burstFrames = frames.InitAbilSlice(321) // Q -> Walk
+	burstFrames[action.ActionAttack] = 297
+	burstFrames[action.ActionAim] = 297
+	burstFrames[action.ActionSkill] = 101
+	burstFrames[action.ActionDash] = 287
+	burstFrames[action.ActionJump] = 285
+	burstFrames[action.ActionSwap] = 283
+}
+
+// Burst attack damage queue generator
+func (c *char) Burst(p map[string]int) action.ActionInfo {
```

Comment:
without extra duration? idk if it's necessary

Reply:
Wdym by extra duration?

Reply:
oh i see canceling the burst using the skill